frontend/react-query/useProject.tsx

```diff
@@ -0,0 +1,57 @@
+import { useQuery, useMutation } from 'react-query';
+import { useRouter } from 'next/router';
+import { PostState, Project } from '@/types/types';
+import { api } from '@/util/api';
+
+type ProjectData = {
+  post_data: Project;
+  post_state: PostState;
+};
+
+export const useProject = () => {
+  const router = useRouter();
+
+  const { isLoading, error, data, refetch } = useQuery<ProjectData, Error>(
+    ['project', router.query.id],
+    () => api(router.asPath).then((res) => res.data)
+  );
+
+  //직군 지원
+  const wantJob = useMutation(
+    (job: string) => api.post(`${router.asPath}/want`, { data: job }),
+    {
+      onSuccess: () => {
+        refetch();
+      },
+    }
+  );
+
+  //지원한 직군 취소
+  const cancleJob = useMutation(
+    (job: string) => api.post(`${router.asPath}/cancle`, { data: job }),
+    {
+      onSuccess: () => {
+        refetch();
+      },
+    }
+  );
```

Comment:
하나의 함수로 관리 할 수 있을 것 같습니다.
```suggestion
  const updateJob = useMutation(
    ({ job, update }: { job: string; update: string }) =>
      api.post(`${router.asPath}/${update}`, { data: job }),
    {
      onSuccess: () => {
        refetch();
      },
    }
  );
```

Reply:
리팩토링 하였습니다!